app/src/main/java/com/ciandt/dragonfly/example/features/feedback/FeedbackActivity.kt

```diff
@@ -123,10 +173,157 @@ class FeedbackActivity : BaseActivity(), FeedbackContract.View {
+                .alpha(0.0f)
+                .setDuration(HIDE_SHOW_ANIMATION_DURATION)
+                .setListener(object : AnimatorListenerAdapter() {
+                    override fun onAnimationEnd(animation: Animator?) {
+                        super.onAnimationEnd(animation)
+                        arrow.setImageDrawable(ContextCompat.getDrawable(this@FeedbackActivity, R.drawable.ic_arrow_down))
+                        arrow.animate()
+                                .alpha(1.0f)
+                                .setDuration(HIDE_SHOW_ANIMATION_DURATION)
+                                .setListener(null)
+
+                    }
+                })
+
+        chipsContainer.visibility = View.VISIBLE
+    }
+
+    private fun collapseResults() {
+
+        toggleButton.isClickable = false
+        toggleButton.animate()
+                .alpha(0.0f)
+                .setDuration(HIDE_SHOW_ANIMATION_DURATION)
+                .setListener(object : AnimatorListenerAdapter() {
+                    override fun onAnimationEnd(animation: Animator?) {
+                        super.onAnimationEnd(animation)
+                        toggleButton.text = getString(R.string.feedback_more_info)
+                        toggleButton.setTextColor(ContextCompat.getColor(this@FeedbackActivity, R.color.feedback_more_info))
+                        toggleButton.animate()
+                                .alpha(1.0f)
+                                .setDuration(HIDE_SHOW_ANIMATION_DURATION)
+                                .setListener(object : AnimatorListenerAdapter() {
+                                    override fun onAnimationEnd(animation: Animator?) {
+                                        super.onAnimationEnd(animation)
+
+                                        toggleButton.isClickable = true
+                                    }
+                                })
+
+                    }
+                })
+
+        arrow.animate()
+                .alpha(0.0f)
+                .setDuration(HIDE_SHOW_ANIMATION_DURATION)
+                .setListener(object : AnimatorListenerAdapter() {
+                    override fun onAnimationEnd(animation: Animator?) {
+                        super.onAnimationEnd(animation)
+                        arrow.setImageDrawable(ContextCompat.getDrawable(this@FeedbackActivity, R.drawable.ic_arrow_up))
+                        arrow.animate()
+                                .alpha(1.0f)
+                                .setDuration(HIDE_SHOW_ANIMATION_DURATION)
+                                .setListener(null)
+                    }
+                })
+
+        chipsContainer.visibility = View.GONE
+    }
+
+    private fun showFeedbackView() {
+
+        if (feedbackView.alpha > 0) {
+            return
+        }
+
+        feedbackView.isClickable = false
+        feedbackView.animate()
+                .alpha(1.0f)
+                .setDuration(FIRST_APPEAR_ANIMATION_DURATION)
```

Comment:
Only once, when TF returns the classifications. To analyse again, user needs to go back and take another snapshot.